master/buildbot/www/service.py

```diff
@@ -386,6 +392,18 @@ def getUserInfos(self, request):
         session = request.getSession()
         return session.user_info
 
+    def getUserInfosFromSessionOrApiToken(self, request):
+        token = request.getHeader(b'X-API-TOKEN')
+        if token:
+            try:
+                session = jwt.decode(token, self.site.api_token_secret, algorithm=APITOKEN_SECRET_ALGORITHM)
+                return session.user_info
```

Comment:
Is the content of user_info subject to change on update or profile change? Tokens might live a long time; you could restrict the claim to maybe a version number, a unique and constant user identifier (username? database entry id?) , a token identifier (e.g. sequential or random serial) for revocation, and get every other user data from the database after successful authentication, otherwise you might decode Realnames, Mailadresses, etc. that are not up-to-date anymore.